src/libsyncengine/update_detection/file_system_observer/localfilesystemobserverworker.cpp

```diff
@@ -83,9 +83,16 @@ void LocalFileSystemObserverWorker::changesDetected(const std::list<std::pair<st
         _needUpdateTimerStart = std::chrono::steady_clock::now();
 
         OperationType opTypeFromOS = changedItem.second;
-        SyncPath absolutePath = changedItem.first.native();
-        SyncPath relativePath = CommonUtility::relativePath(_syncPal->localPath(), absolutePath);
+        const SyncPath absolutePath = changedItem.first.native();
+        const SyncPath relativePath = CommonUtility::relativePath(_syncPal->localPath(), absolutePath);
 
+        if (_syncPal->isTmpBlacklisted(relativePath, ReplicaSide::Local)) {
+            _syncPal->removeItemFromTmpBlacklist(relativePath);
+            if (opTypeFromOS == OperationType::Edit) {
+                NodeId itemId = _snapshot->itemId(relativePath);
+                if (!itemId.empty()) _snapshot->setLastModified(itemId, 0);
+            }
+        }
```

Comment:
Iterating over each element of the path could be very time-consuming. Could we imagine simply remove the item from the blacklist based on its ID?
Do you use path because we might receive a notification on an item whom one of the ancestor is blacklisted?

Reply:
Optimized here: Done here: [e637b94](https://github.com/Infomaniak/desktop-kDrive/pull/356/commits/e637b94214070ffd08e28dd9945ad086c069a4bf)